backend/press/press.go

```diff
@@ -0,0 +1,1376 @@
+	return o.Object
+}
+
+// Open opens the file for read.  Call Close() on the returned io.ReadCloser. Note that this call requires quite a bit of overhead.
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (rc io.ReadCloser, err error) {
+	err = o.loadMetadataIfNotLoaded(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// If we're uncompressed, just pass this to the underlying object
+	if o.meta.Mode == Uncompressed {
+		return o.Object.Open(ctx, options...)
+	}
+	// Get offset and limit from OpenOptions, pass the rest to the underlying remote
+	var openOptions []fs.OpenOption = []fs.OpenOption{&fs.SeekOption{Offset: 0}}
+	var offset, limit int64 = 0, -1
+	for _, option := range options {
+		switch x := option.(type) {
+		case *fs.SeekOption:
+			offset = x.Offset
+		case *fs.RangeOption:
+			offset, limit = x.Decode(o.Size())
+		default:
+			openOptions = append(openOptions, option)
+		}
+	}
+	// Get a chunkedreader for the wrapped object
+	chunkedReader := chunkedreader.New(ctx, o.Object, initialChunkSize, maxChunkSize)
+	// Get file handle
+	var file io.Reader
+	if offset != 0 {
+		file, err = sgzip.NewReaderAt(chunkedReader, &o.meta.CompressionMetadata, offset)
+	} else {
+		file, err = sgzip.NewReader(chunkedReader)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var fileReader io.Reader
+	if limit != -1 {
+		fileReader = io.LimitReader(file, limit)
+	} else {
+		fileReader = file
+	}
+	// Return a ReadCloser
+	return combineReaderAndCloser(fileReader, chunkedReader), nil
+}
+
+// ObjectInfo describes a wrapped fs.ObjectInfo for being the source
+type ObjectInfo struct {
+	src    fs.ObjectInfo
+	fs     *Fs
+	remote string
+	size   int64
+}
+
+func (f *Fs) wrapInfo(src fs.ObjectInfo, newRemote string, size int64) *ObjectInfo {
+	return &ObjectInfo{
+		src:    src,
+		fs:     f,
+		remote: newRemote,
+		size:   size,
+	}
+}
+
+// Fs returns read only access to the Fs that this object is part of
+func (o *ObjectInfo) Fs() fs.Info {
+	if o.fs == nil {
+		panic("stub ObjectInfo")
+	}
+	return o.fs
+}
+
+// String returns string representation
+func (o *ObjectInfo) String() string {
+	return o.src.String()
+}
+
+// Storable returns whether object is storable
+func (o *ObjectInfo) Storable() bool {
+	return o.src.Storable()
+}
+
+// Remote returns the remote path
+func (o *ObjectInfo) Remote() string {
+	if o.remote != "" {
+		return o.remote
+	}
+	return o.src.Remote()
+}
+
+// Size returns the size of the file
+func (o *ObjectInfo) Size() int64 {
+	if o.size != -1 {
+		return o.size
+	}
+	return o.src.Size()
+}
+
+// ModTime returns the modification time
+func (o *ObjectInfo) ModTime(ctx context.Context) time.Time {
+	return o.src.ModTime(ctx)
+}
+
+// Hash returns the selected checksum of the file
+// If no checksum is available it returns ""
+func (o *ObjectInfo) Hash(ctx context.Context, ht hash.Type) (string, error) {
+	if ht != hash.MD5 {
+		return "", hash.ErrUnsupported
+	}
+	if o.Size() != o.src.Size() {
```

Comment:
I don't understand the logic here?

Reply:
No idea either. I don't think that was intentional.